### ISSUE TITLE
Kotlin: improve dev experience for KtLint failures

### ISSUE THREAD

**Issue by david-allison-1** [ankidroid/Anki-Android#9019]
* The error in CI doesn't specify how to fix this
* Could we use a pre-commit hook?

```

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':AnkiDroid:ktlintMainSourceSetCheck'.
> A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.ConsoleReportWorkAction
   > KtLint found code style violations. Please see the following reports:
     - /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/build/reports/ktlint/ktlintMainSourceSetCheck/ktlintMainSourceSetCheck.txt

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

**Comment by mikehardy** [comment]
pre-commit hooks are fantastic IMHO. At least for the linting I'm a huge fan, I understand some may not want to run unit tests or emulator tests each time, but linting at least should not be too painful

**Comment by ShridharGoel** [comment]
Should we use a pre-commit hook for checking or for formatting?

**Comment by david-allison-1** [comment]
Sounds good to me

**Comment by david-allison-1** [comment]
Fixed in  #9028
